presentation/src/main/java/com/activity_sync/presentation/presenters/ChangePasswordPresenter.java

```diff
@@ -0,0 +1,83 @@
+package com.activity_sync.presentation.presenters;
+
+import com.activity_sync.presentation.utils.StringUtils;
+import com.activity_sync.presentation.views.IChangePasswordView;
+
+import rx.Scheduler;
+
+public class ChangePasswordPresenter extends Presenter<IChangePasswordView>
+{
+    private Scheduler uiThread;
+
+    public ChangePasswordPresenter(IChangePasswordView view, Scheduler uiThread)
+    {
+        super(view);
+        this.uiThread = uiThread;
+    }
+
+    @Override
+    public void start()
+    {
+        super.start();
+
+        subscriptions.add(view.onSaveClick()
+                .observeOn(uiThread)
+                .subscribe(o -> {
+                    boolean canContinue = true;
+
```

Comment:
3 things:

Putting more that one condition directly in if statement is not a good practice. You should create here something like this: you should put that conditions in a boolean method with human readable name.

Secondly, I think that checking nulls should be at the beginning

Thirdly, I am not sure if that "if" inside "if" is good. It looks strange.